Fixing issue with correcting escaped double quotes

ykuijs opened the pull request:
This PR fixes an issue where escaped double quotes are not corrected.

FabienTschanz commented:
@ykuijs Do you have an example what your PR fixes? It's a bit hard to understand just from the code 😄

ykuijs commented:
The has been a multi stage problem:

The issue I initially wanted to fix was that the tenant name was not converted into the generic $OrganizationName variable. It still had the tenant specific onmicrosoft.com values in the output:
```PowerShell
AADAdministrativeUnit "AADAdministrativeUnit-London Office"
{
	ApplicationId                = $ConfigurationData.NonNodeData.ApplicationId;
	ApplicationSecret            = New-Object System.Management.Automation.PSCredential ('ApplicationSecret', (ConvertTo-SecureString $ConfigurationData.NonNodeData.ApplicationSecret -AsPlainText -Force));
	DisplayName                  = "London Office";
	Ensure                       = "Present";
	Id                           = "8d2cd067-a70f-4f58-aa57-231ae32c0d7d";
	IsMemberManagementRestricted = $False;
	Members                      = @(
		MSFT_MicrosoftGraphMember{
			Type = 'Group'
			Identity = 'IT Helpdesk Team'
		}
	);
	ScopedRoleMembers            = @(
		MSFT_MicrosoftGraphScopedRoleMembership{
			RoleName = 'Helpdesk Administrator'
			RoleMemberInfo = MSFT_MicrosoftGraphMember{
				Type = 'User'
				Identity = 'AlexW@M365x648977.OnMicrosoft.com'
			}

		}
		MSFT_MicrosoftGraphScopedRoleMembership{
			RoleName = 'Helpdesk Administrator'
			RoleMemberInfo = MSFT_MicrosoftGraphMember{
				Type = 'User'
				Identity = 'AllanD@M365x648977.OnMicrosoft.com'
			}

		}
		MSFT_MicrosoftGraphScopedRoleMembership{
			RoleName = 'Helpdesk Administrator'
			RoleMemberInfo = MSFT_MicrosoftGraphMember{
				Type = 'User'
				Identity = 'AdeleV@M365x648977.OnMicrosoft.com'
			}
		}
	);
	TenantId                     = $OrganizationName;
}
```

After fixing that, I discovered that in CIM instances the parameter values were written with a single quote which will not work with variables:
```PowerShell
MSFT_MicrosoftGraphScopedRoleMembership{
	RoleName = 'Helpdesk Administrator'
	RoleMemberInfo = MSFT_MicrosoftGraphMember{
		Type = 'User'
		Identity = 'AdeleV@$OrganizationName'
	}
```

So I had to convert the single quote into double quotes. But when using CIM instances, this resulted in escaped double quotes since they were inserted into a string. The code did not handle that properly, which resulted in the double quotes to be removed, leaving the back ticks:
```PowerShell
MSFT_MicrosoftGraphScopedRoleMembership{
	RoleName = `Helpdesk Administrator`
	RoleMemberInfo = MSFT_MicrosoftGraphMember{
		Type = `User`
		Identity = `AdeleV@M365x648977.OnMicrosoft.com`
	}
```

My first update fixed that issue for CIM instance objects that were used in an array. But then we discovered that CIM instances not in an array (so just single objects) still had this issue. My PR of yesterday fixed those. The `Convert-DSCStringParamToVariable` function however had a IsCIMArray parameter and it didn't make sense to set that to true when the item isn't an array. Renaming the parameter would be a breaking change. That is why I added a IsCIMObject parameter which basically does the same as the IsCIMArray parameter.

The escaped double quotes strings are now converted back to double quoted strings, so the export is created properly and also converts the tenant name into the OrganizationName variable.

FabienTschanz commented:
@ykuijs I see, thank you for explaining. The issue now is that double quotes inside of a string in the CIM instance won't be escaped. I'll take a look at that.
